core/apple/test/SourceNSInputStreamTest.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENCE file.
+ */
+
+package kotlinx.io
+
+import kotlinx.cinterop.*
+import platform.Foundation.NSInputStream
+import platform.Foundation.NSStreamStatusClosed
+import platform.Foundation.NSStreamStatusNotOpen
+import platform.Foundation.NSStreamStatusOpen
+import platform.darwin.NSUIntegerVar
+import platform.darwin.UInt8Var
+import kotlin.test.*
+
+@OptIn(UnsafeNumber::class)
+class SourceNSInputStreamTest {
+    @Test
+    fun bufferInputStream() {
+        val source = Buffer()
+        source.writeString("abc")
```

Comment:
Please add some cases with longer data written into the buffer to cover the case when multiple segments are involved in reading from a source.

Reply:
I added these longer input data test cases: https://github.com/Kotlin/kotlinx-io/pull/174/commits/365c35421ce4402d5127abc3fca6478edb5b54cd